R/internal.R

```diff
@@ -33,7 +33,41 @@ coalesceifnull <- function(x, y) {
 #' @importFrom reticulate import
 boto <- reticulate::import("boto", delay_load = TRUE)
 
+read.text = function(pathname) {
+  if (file.exists(pathname)) {
+	      return (paste(readLines(pathname), collapse="\n"))
+  } 
+}
+
+check_aws_credentials <- function() {
+  aws_credentials_path = "~/.aws/credentials"
+  aws_configure_path = "~/.aws/configure"
+  renviron_path = "/etc/R/Renviron.site"
+  rprofile_path = ".Rprofile"
+  aws_key_env_set = Sys.getenv("AWS_ACCESS_KEY_ID") != ""
+  aws_configure_exists = file.exists(aws_configure_path) 
+  aws_credentials_exists = file.exists(aws_credentials_path)
+  # Does not need checking as both would set env var, keeping it here for completness
+  renviron_exists = file.exists(renviron_path)
+  rprofile_exists = file.exists(rprofile_path)
+  if (aws_key_env_set) {
+    log_if_verbose("AWS_ACCESS_KEY_ID is set, it will override other aws credentials") 	
+  } else if (aws_credentials_exists) {
+    if (grepl("aws_access_key_id", read.text(aws_credentials_path))) {
```

Comment:
I would turn this and the one below it into `else if (aws_credentials_exists && grepl(...))` or whatever the `and` syntax is in R, but that's only style and it might make the line too long for your taste.